src/site/content/en/blog/sanitizer/index.md

```diff
@@ -78,32 +78,14 @@ const $div = document.querySelector('div')
 const user_input = `<em>hello world</em><img src="" onerror=alert(0)>`
 const sanitizer = new Sanitizer()
 $div.setHTML(user_input, sanitizer) // <div><em>hello world</em><img src=""></div>
-
 ```
 
 It is worth noting that `setHTML()` is defined on [`Element`](https://developer.mozilla.org/docs/Web/API/Element). Being a method of `Element`, the context to parse is self-explanatory (`<div>` in this case), the parsing is done once internally, and the result is directly expanded into the DOM.
 
-If you don't want to expand directly into the DOM, you can also get the result as an `HTMLElement`.
-
+To get the result of sanitization as a string, you can use `.innerHTML` from the `setHTML()` results.
 
 ```js
-const user_input = `<em>hello world</em><img src="" onerror=alert(0)>`
-const sanitizer = new Sanitizer()
-sanitizer.sanitizeFor("div", user_input) // HTMLDivElement <div>
-```
-
-If you want to get the result of sanitization as a string, you can use `.innerHTML` from the `HTMLElement` you got here. (You still need to specify the appropriate context to parse.)
-
-
-```js
-sanitizer.sanitizeFor("div", user_input).innerHTML // <em>hello world</em><img src="">
-```
-
-If you already have a user-controlled `DocumentFragment` and you want to remove only the harmful stuff from it, use `sanitize()`.
-
-
-```js
-$div.replaceChildren(s.sanitize($userDiv));
+$div.setHTML(user_input, sanitizer).innerHTML // <em>hello world</em><img src="">
```

Comment:
This shouldn't quite work, because .setHTML returns void. There's also one more thing (that I think is important): If $div is connected, that is, if it's part of the main document, then inserting it would trigger some actions. For example, it would start loading images.

I think the concept is important, though. I'd describe this as follows:

If you don't want to expand directly into the DOM, you can create a new element and sanitize into that.
const div = document.createElement("div");
div.setHTML(user_input, sanitizer);
return div.innerHTML;

This has the advantage the $div is definitely not connected to the DOM, and thus neither loads images (or other content), and doesn't have active event handlers.

-----------------

There is some ongoing discussion whether we should improve the API to allow this directly. But as it presently stands, this is probably the best way to do it.

One interesting thing - which I'm just dropping here, even though it doesn't directly apply to this issue - is a paper that was recently published on sanitizer use in HTML (https://swag.cispa.saarland/papers/klein2022hand.pdf). I think it's interesting because it gives us more information about the audience of the Sanitizer API, and hence this article: They found that many sites "sanitize" their input using string operations (.replace, or regexp matching), and often do it wrong. IMHO, these people are our primary audience, and I hope it pays off to be a bit more clean on how we think this should be used.